### PR TITLE
WIP Use Rubocop

### LINKED ISSUE #564: [Maintenance] Install RuboCop (Ruby Style Analyzer)
At the moment we have different ruby code syntax and styles (e.g. `:variable => 'string'` vs. `variable: 'string'`) in wheelmap. Using [rubocop](https://github.com/bbatsov/rubocop) could help us to bring our different styles aligned and increase our code quality by refactoring. Since establishing this gem may also cause a lot of error alerts (travis), we should calculcate a bit of time for this enhancement.

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes #564 

Still some work to do. :)

### CODE REVIEW

File: app/models/poi.rb [hunk 1]
```diff
@@ -426,26 +421,26 @@ def distance_to_in_meter(other_poi)
...
-  def self.distance_search(search_string, bbox, limit=200)
+  def self.distance_search(search_string, bbox, limit = 200)
     @node_ids = []
     left, bottom, right, top = bbox.split(',').map(&:to_f)
     @bbox = Bbox.new(left, bottom, right, top)
     max_box = Bbox.max
     # Increase the bounding box as long enough nodes are found or it covers the whole world
-    while @node_ids.size <= limit and @bbox.within?(max_box) do
-      found_node_ids = self.select(:osm_id).search_scope(search_string).within_bbox(@bbox.west, @bbox.south, @bbox.east, @bbox.north).map(&:id)
+    while (@node_ids.size <= limit) && @bbox.within?(max_box)
```
Comment (Hoverbear) [inline app/models/poi.rb]: Maybe a `do` missing

File: config/initializers/carrierwave.rb [hunk 2]
```diff
@@ -57,7 +55,7 @@ def effectively_resize_to_limit(width, height, format = 'jpg', quality = 70)
 end
 
 CarrierWave.configure do |config|
-  config.permissions = 0644
-#  config.directory_permissions = 0755
+  config.permissions = 0o644
```
Comment (Hoverbear) [inline config/initializers/carrierwave.rb]: Should this be octal?

File: config/initializers/wheelmap.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # encoding: UTF-8
-WheelchairStatus = %w{yes limited no unknown}
+WheelchairStatus = [%w(yes limited no unknown)].freeze
```
Comment (Hoverbear) [inline config/initializers/wheelmap.rb]: Check this.

File: lib/capistrano/tasks/console.rake [hunk 4]
```diff
@@ -16,6 +16,6 @@ namespace :rails do
 
   def run_interactively(command, user)
     info "Running `#{command}` as #{user}@#{host}"
-    exec %Q(ssh #{user}@#{host} -p #{fetch(:ssh_options)[:port] || 22 } -i #{fetch(:ssh_options)[:keys].join(' ')} -t "bash --login -c 'cd #{fetch(:deploy_to)}/current && #{command}'")
+    exec %(ssh #{user}@#{host} -p #{fetch(:ssh_options)[:port] || 22} -i #{fetch(:ssh_options)[:keys].join(' ')} -t "bash --login -c 'cd #{fetch(:deploy_to)}/current && #{command}'")
```
Comment (Hoverbear) [inline lib/capistrano/tasks/console.rake]: Need to check

File: lib/tasks/export.rake [hunk 5]
```diff
@@ -316,13 +325,11 @@ namespace :export do
     end
   end
 
-
-  task :tags => :environment do
-    puts "Tags = {"
+  task tags: :environment do
+    puts 'Tags = {'
```
Comment (Hoverbear) [inline lib/tasks/export.rake]: Seriously!?

File: lib/tasks/housekeeping.rake [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
-
   desc 'Remove queued nodes from osm file'
-  task :remove_queued_nodes => :environment do
+  task remove_queued_nodes: :environment do
     file = ENV['file']
-    raise "Run rake housekeeping:remove_queued_nodes file=<yourfile>" unless file
+    raise 'Run rake housekeeping:remove_queued_nodes file=<yourfile>' unless file
     File.open(file, 'r') do |f|
-      while line = f.readline do
+      while (line = f.readline)
```
Comment (Hoverbear) [inline lib/tasks/housekeeping.rake]: Another new is gone

File: spec/mailers/user_mailer_spec.rb [hunk 7]
```diff
@@ -1,25 +1,22 @@
 # encoding: UTF-8
-require "spec_helper"
+require 'spec_helper'
 
 describe UserMailer do
-
   let :user do
-    FactoryGirl.create(:user, email: 'chuck@heartbreaker.com' )
+    FactoryGirl.create(:user, email: 'chuck@heartbreaker.com')
```
Comment (Hoverbear) [inline spec/mailers/user_mailer_spec.rb]: Awesome email

File: config/initializers/wheelmap.rb (hunk 3 above)
Comment (Hoverbear) [inline config/initializers/wheelmap.rb]: This is wrong

File: spec/mailers/user_mailer_spec.rb (hunk 7 above)
Comment (Xylakant) [inline spec/mailers/user_mailer_spec.rb]: not cool:

´´´´
 felix.gilcher@redemption-space  ~  whois hearbreaker.com
[Querying whois.verisign-grs.com]
[Redirected to whois.godaddy.com]
[Querying whois.godaddy.com]
[whois.godaddy.com]
Domain Name: HEARBREAKER.COM
Registry Domain ID: 1906059810_DOMAIN_COM-VRSN
Registrar WHOIS Server: whois.godaddy.com
Registrar URL: http://www.godaddy.com
Update Date: 2015-04-15T12:25:42Z
Creation Date: 2015-02-28T01:11:09Z
Registrar Registration Expiration Date: 2017-02-28T01:11:09Z
Registrar: GoDaddy.com, LLC
Registrar IANA ID: 146
Registrar Abuse Contact Email: abuse@godaddy.com
Registrar Abuse Contact Phone: +1.4806242505
Domain Status: clientTransferProhibited http://www.icann.org/epp#clientTransferProhibited
Domain Status: clientUpdateProhibited http://www.icann.org/epp#clientUpdateProhibited
Domain Status: clientRenewProhibited http://www.icann.org/epp#clientRenewProhibited
Domain Status: clientDeleteProhibited http://www.icann.org/epp#clientDeleteProhibited
Registry Registrant ID: Not Available From Registry
´´´´

Domain is registered.

Comment (Hoverbear) [inline spec/mailers/user_mailer_spec.rb]: Issue created https://github.com/sozialhelden/wheelmap/issues/584

**Pull request merged**